src/common/gpio_servo.rs

```diff
@@ -0,0 +1,195 @@
+//! A generalized servo implementation that uses a PWM signal on a GPIO pin
+//! via methods on a board component
+//!
+//! # Creating a GPIO pin servo and turning it 90 degrees
+//!
+//! ```ignore
+//!
+//! let board = FakeBoard::new(vec![]);
+//!
+//! let servo_settings = GpioServoSettings {
+//!     min_angle_deg: 0,
+//!     max_angle_deg: 180,
+//!     min_period_us: 500,
+//!     max_period_us: 2500,
+//!     pwm_resolution: 0
+//! }
+//!
+//! let mut servo = GpioServo::new(board, 12, servo_settings);
+//!
+//! servo.move_to(90).unwrap()
+//!
+//! ```
+
+use std::sync::{Arc, Mutex};
+
+use super::{
+    actuator::Actuator,
+    board::{Board, BoardType},
+    config::{AttributeError, ConfigType},
+    registry::{get_board_from_dependencies, ComponentRegistry, Dependency},
+    servo::{Servo, ServoType},
+    status::Status,
+};
+
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
+    if registry.register_servo("gpio", &from_config).is_err() {
```

Comment:
it happens automatically but where it happens depends on which part of the namespace we're talking about. This will have the API namespace "rdk:component:servo" and the model namespace "rdk:builtin:gpio". The "rdk" part of the API namespace is implicitly set in the component config's "namespace" key by app (this is temporary until the work for the new robot details page gets underway, long story that I can explain separately). The "rdk:builtin" portion is also prefixed to the model name by app. "servo" is added by the `register_servo` function and the `insert_resources` function on robot depending on whether you're registering the constructor or the actual component

Reply:
OK. I need to think about what the implications of that are for modular resources. Thanks for providing the details.